js/src/Zone.js

```diff
@@ -40,51 +40,28 @@ export default class Zone extends UXBase {
 	marks = this.props.marks.map(markName => {
 		const mark = ExpectedMark.create(markName);
 
-		mark.onComplete(mark => {
-			// Call Zone's `onMark` callback
-			this.props.onMark(mark.name);
-
-			if (this.checkCompletion()) {
-				this.complete(mark);
+		mark.onComplete(completeMark => {
+			// pass the event upstream
+			this.props.onMark(markName);
+			if (this.marks.every(m => m.marked)) {
+				this.measure(markName);
 			}
 		});
 
 		return mark;
 	});
 
-	/**
-	 * Check if all marks for the zone have already been recorded
-	 */
-	checkCompletion() {
-		if (
-			typeof window.performance === 'undefined' ||
-			typeof window.performance.getEntriesByType === 'undefined'
-		) {
-			return false;
-		}
-
-		const recordedMarks = window.performance.getEntriesByType('mark');
-
-		return this.marks.every(mark =>
-			recordedMarks.find(recordedMark => recordedMark.name === mark.name)
-		);
-	}
-
 	/**
 	 * Records measure on Performance Timeline and calls onMeasure callback
 	 *
 	 * @param {ExpectedMark} lastMark last mark that triggered completion
 	 */
-	complete(lastMark) {
+	measure(endMarkName) {
```

Comment:
renamed this method to make it a little more clear what it's doing and to keep the naming convention parallel with `.mark`